pkg/client/record/event.go

```diff
@@ -51,16 +51,17 @@ type EventRecorder interface {
 	// Event constructs an event from the given information and puts it in the queue for sending.
 	// 'object' is the object this event is about. Event will make a reference-- or you may also
 	// pass a reference to the object directly.
+	// 'timestamp' refers to the time at which the event was originally observed or generated.
 	// 'reason' is the reason this event is generated. 'reason' should be short and unique; it will
 	// be used to automate handling of events, so imagine people writing switch statements to
 	// handle them. You want to make that easy.
 	// 'message' is intended to be human readable.
 	//
 	// The resulting event will be created in the same namespace as the reference object.
-	Event(object runtime.Object, reason, message string)
+	Event(object runtime.Object, timestamp util.Time, reason, message string)
```

Comment:
The event's timestamp is expected to correlate with when the actual event occurred. Event* calls not taking timestamp was fine all along since all the events were generated by kube components directly. Only now we are starting to proxy events. I would argue that the original interface should have taken timestamps.